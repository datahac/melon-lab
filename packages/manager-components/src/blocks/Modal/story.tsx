import { storiesOf } from '@storybook/react';
import React from 'react';
import Modal from './index';

storiesOf('Blocks|Modal', module).add('Default', () => {
  return (
    <Modal isOpen loading title="Hello World">
      Hello World
    </Modal>
  );
});
