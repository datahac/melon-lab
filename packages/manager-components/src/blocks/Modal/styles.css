:global(.modal__wrap) {
  background: rgb(255, 253, 243);
  position: absolute;
  border: none;
  overflow: auto;
  border-radius: 0px;
  outline: none;
  left: 16px;
  right: 16px;
  bottom: 16px;
  top: 16px;
}

@media (--viewport-m) {
  :global(.modal__wrap) {
    top: 40px;
    width: 700px;
    bottom: auto;
    left: auto;
    right: auto;
  }
}

:global(.modal__overlay) {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.8);
  z-index: 200;
  display: flex;
  justify-content: center;
}

:global(.modal__actions) {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

:global(.modal__action) {
  flex: 0 0 40%;
}

:global(.modal__spinner) {
  width: 50px;
  height: 50px;
  margin: var(--space-s) auto;
}

.modal__title {
  font-weight: var(--font-weight-bold);
  font-size: var(--font-size-l);
  background-color: var(--color-secondary);
  margin-top: 0;
  padding: var(--space-s) var(--space-m);
}

.modal__content {
  margin: var(--space-m);
}
