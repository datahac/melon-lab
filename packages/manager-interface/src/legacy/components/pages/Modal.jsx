import React from 'react';
import ModalComponent from '@melonproject/manager-components/blocks/Modal';
import PasswordForm from '../../containers/PasswordForm';
import FeeForm from '../../containers/FeeForm';

export const types = {
  ERROR: 'ERROR',
  LOADING: 'LOADING',
  CONFIRM: 'CONFIRM',
  PASSWORD: 'PASSWORD',
  INFO: 'INFO',
};

const Modal = ({
  isOpen,
  type,
  title,
  body,
  fees,
  method,
  primaryInteraction,
  secondaryInteraction,
  interactionHandler,
  handleSubmit,
  onAfterOpen,
}) => {
  return (
    <ModalComponent
      isOpen={isOpen}
      title={title}
      body={body}
      primaryInteraction={primaryInteraction}
      secondaryInteraction={secondaryInteraction}
      interactionHandler={interactionHandler}
      handleSubmit={handleSubmit}
      onAfterOpen={onAfterOpen}
      loading={type === types.LOADING}
      error={type === types.ERROR}
    >
      {method &&
        type !== types.ERROR && (
          <div>
            <p>
              The following method on the Melon Smart Contracts will be
              executed:
            </p>
            <p>
              <b>{method}</b>
            </p>
          </div>
        )}

      {fees && <FeeForm fees={fees} />}
      {type === types.PASSWORD && <PasswordForm />}
    </ModalComponent>
  );
};

export default Modal;
