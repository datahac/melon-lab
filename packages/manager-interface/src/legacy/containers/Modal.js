import { connect } from 'react-redux';
import Modal from '../components/pages/Modal';
import { actions, interactions } from '../actions/modal';
import { getEnvironment, setEnvironment } from '@melonproject/melon.js';

const mapStateToProps = state => ({
  ...state.modal,
  type: state.modal.modalType,
});

const mapDispatchToProps = dispatch => ({
  interactionHandler: (event, interaction) => {
    // Hint: Submit is handled by the form, not as action. But we need to
    // prevent form submitting for the other actions.
    if (interaction === interactions.CANCEL) {
      dispatch(actions.cancel());
    } else {
      dispatch(actions.confirmed());
    }
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Modal);
