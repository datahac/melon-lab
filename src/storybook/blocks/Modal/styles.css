:global(.modal__wrap) {
  background: rgb(255, 253, 243);
  position: absolute;
  border: none;
  overflow: auto;
  border-radius: 0px;
  outline: none;
  left: 16px;
  right: 16px;
  bottom: 16px;
  top: 16px;
}

@media (--viewport-m) {
  :global(.modal__wrap) {
    top: 40px;
    width: 700px;
    bottom: auto;
    left: auto;
    right: auto;
  }
}

:global(.modal__overlay) {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.8);
  z-index: 200;
  display: flex;
  justify-content: center;
}

.modal__actions {
  margin-top: var(--space-m);
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.modal__action {
  flex: 0 0 40%;
}

.modal__action:first-child:nth-last-child(1) {
  flex: 0 0 100%;
}

.modal__action:first-child:nth-last-child(2),
.modal__action:first-child:nth-last-child(2) ~ li {
  flex: 0 0 40%;
}

.modal__title {
  font-weight: var(--font-weight-bold);
  font-size: var(--font-size-l);
  background-color: var(--color-secondary);
  margin-top: 0;
  padding: var(--space-s) var(--space-m);
}

.modal__content {
  margin: var(--space-m);
}

.modal__spinner {
  margin: var(--space-m) auto;
  width: 50px;
}
