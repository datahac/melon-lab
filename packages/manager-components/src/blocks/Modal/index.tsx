import React, { Fragment, StatelessComponent } from 'react';
import ReactModal from 'react-modal';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Spinner from '~/blocks/Spinner';

import styles from './styles.css';

export interface ModalProps {
  isOpen?: boolean;
  title?: string;
  body?: string;
  primaryInteraction?: () => void;
  secondaryInteraction?: () => void;
  interactionHandler?: () => void;
  onAfterOpen?: () => void;
  handleSubmit?: () => void;
  loading?: boolean;
  error?: boolean;
}

const renderButtons = (
  primaryInteraction,
  secondaryInteraction,
  interactionHandler,
) => {
  const onClickPrimaryInteraction = event => {
    interactionHandler(event, primaryInteraction);
  };

  const onClickSecondaryInteraction = event => {
    interactionHandler(event, secondaryInteraction);
  };

  return secondaryInteraction ? (
    <div className="modal__actions">
      <div className="modal__action">
        <Button
          type="button"
          style="secondary"
          onClick={onClickSecondaryInteraction}
        >
          {secondaryInteraction}
        </Button>
      </div>
      <div className="modal__action">
        <Button type="submit" onClick={onClickPrimaryInteraction}>
          {primaryInteraction}
        </Button>
      </div>
    </div>
  ) : (
    primaryInteraction && (
      <Button type="submit" onClick={onClickPrimaryInteraction}>
        {primaryInteraction}
      </Button>
    )
  );
};

const Modal: StatelessComponent<ModalProps> = ({
  isOpen,
  title,
  body,
  primaryInteraction,
  secondaryInteraction,
  interactionHandler,
  onAfterOpen,
  children,
  loading,
  error,
  handleSubmit,
}) => {
  return (
    <ReactModal
      className="modal__wrap"
      overlayClassName="modal__overlay"
      onAfterOpen={onAfterOpen}
      isOpen={isOpen}
    >
      <style jsx>{styles}</style>
      <div className="modal__title">{title}</div>
      <div className="modal__content">
        <p>{body}</p>
        {children}

        {!loading ? (
          <Form onSubmit={handleSubmit}>
            {renderButtons(
              primaryInteraction,
              secondaryInteraction,
              interactionHandler,
            )}
          </Form>
        ) : (
          <div className="modal__spinner">
            <Spinner size="inflated" />
          </div>
        )}
      </div>
    </ReactModal>
  );
};

export default Modal;
