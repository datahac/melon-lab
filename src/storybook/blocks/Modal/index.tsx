import React, { StatelessComponent, Fragment } from 'react';
import ReactModal from 'react-modal';

import styles from './styles.css';

export interface ModalProps {
  title?: string;
  text?: string;
  PrimaryAction;
  PrimaryActionProps;
  SecondaryAction;
  SecondaryActionProps;
  ContentWrapper;
  ContentWrapperProps;
}

const Modal: StatelessComponent<ModalProps> = ({
  PrimaryAction,
  PrimaryActionProps = {},
  SecondaryAction,
  SecondaryActionProps = {},
  ContentWrapper,
  ContentWrapperProps = {},
  children,
  title,
  text,
  ...props
}) => (
  <ReactModal
    className="modal__wrap"
    overlayClassName="modal__overlay"
    ariaHideApp={false}
    {...props}
  >
    <style jsx>{styles}</style>
    <div className="modal__title">{title}</div>
    <div className="modal__content">
      {ContentWrapper ? (
        <ContentWrapper {...ContentWrapperProps}>
          {text && <p>{text}</p>}
          {children}

          <div className="modal__actions">
            {PrimaryAction && (
              <div className="modal__action">
                <PrimaryAction {...PrimaryActionProps} />
              </div>
            )}

            {SecondaryAction && (
              <div className="modal__action">
                <SecondaryAction {...SecondaryActionProps} />
              </div>
            )}
          </div>
        </ContentWrapper>
      ) : (
        <Fragment>
          {text && <p>{text}</p>}
          {children}

          <div className="modal__actions">
            {PrimaryAction && (
              <div className="modal__action">
                <PrimaryAction {...PrimaryActionProps} />
              </div>
            )}

            {SecondaryAction && (
              <div className="modal__action">
                <SecondaryAction {...SecondaryActionProps} />
              </div>
            )}
          </div>
        </Fragment>
      )}
    </div>
  </ReactModal>
);

export default Modal;
