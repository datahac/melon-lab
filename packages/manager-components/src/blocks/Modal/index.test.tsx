import React from 'react';
import Modal from './index';

describe('Modal', () => {
  const defaultElement = (
    <Modal isOpen title="Hello World">
      Hello World
    </Modal>
  );

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
