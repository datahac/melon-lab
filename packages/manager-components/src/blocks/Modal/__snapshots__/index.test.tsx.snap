// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal__wrap"
  closeTimeoutMS={0}
  isOpen={true}
  overlayClassName="modal__overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash modal__title"
  >
    Hello World
  </div>
  <div
    className="jsx-__scopedHash modal__content"
  >
    <p
      className="jsx-__scopedHash"
    />
    Hello World
    <Form />
  </div>
</Modal>
`;
